pkg/gather/clusterconfig/clusterconfig.go

```diff
@@ -18,38 +21,67 @@ import (
 	"github.com/openshift/insights-operator/pkg/record"
 )
 
-var serializer = scheme.Codecs.LegacyCodec(configv1.SchemeGroupVersion)
+var (
+	serializer     = scheme.Codecs.LegacyCodec(configv1.SchemeGroupVersion)
+	kubeSerializer = kubescheme.Codecs.LegacyCodec(corev1.SchemeGroupVersion)
+)
 
 type Gatherer struct {
-	client configv1client.ConfigV1Interface
+	client     configv1client.ConfigV1Interface
+	coreClient corev1client.CoreV1Interface
 
 	lock        sync.Mutex
 	lastVersion *configv1.ClusterVersion
 }
 
-func New(client configv1client.ConfigV1Interface) *Gatherer {
+func New(client configv1client.ConfigV1Interface, coreClient corev1client.CoreV1Interface) *Gatherer {
 	return &Gatherer{
-		client: client,
+		client:     client,
+		coreClient: coreClient,
 	}
 }
 
 var reInvalidUIDCharacter = regexp.MustCompile(`[^a-z0-9\-]`)
 
 func (i *Gatherer) Gather(ctx context.Context, recorder record.Interface) error {
-	return collectRecords(ctx, recorder,
-		func() ([]record.Record, []error) {
-			config, err := i.client.ClusterOperators().List(metav1.ListOptions{})
-			if err != nil {
-				return nil, []error{err}
-			}
-			records := make([]record.Record, 0, len(config.Items))
-			for i := range config.Items {
-				records = append(records, record.Record{Name: fmt.Sprintf("config/clusteroperator/%s", config.Items[i].Name), Item: ClusterOperatorAnonymizer{&config.Items[i]}})
-			}
-			return records, nil
-		},
+	return record.Collect(ctx, recorder,
+		record.Aggregate(
```

Comment:
Why do we record.Aggregate() the first 2 things, and just comma list the others things?

Reply:
Can do it in a follow up - agree it's not as clean as it should be

Reply:
(probably would be clearer if the entire thing was `fns ... func() ([]Record, []error)`